### DISCUSSION TITLE
Possible to create checkpoints / images during the training?

### DISCUSSION THREAD

**Discussion by quasiblob** [bmaltais/kohya_ss#201] [Q&A]
It feels a bit restrictive to not be able to see the training progress. 

Also, like with Automatic1111's Textual Inversion training, it would be nice to be able to save checkpoints after x amount of steps.

**Comment by vivi-gomez** [comment]
I am not sure but there is an option "Save every N epochs".
Although in webgui1111 it is much more clear to save even when canceling

**Reply by quasiblob** [reply]
Thanks! 

That sounds good for saving intermediate models, but is there also an option to somehow render test images to see if training is going OK?

I have to admit I haven't yet had time to check all the available options. I have a hunch it might not possible as this GUI doesn't have any txt2img anyway.

**Reply by runner22k** [reply]
You can try tensorboard to monitor logs to understand the learning rate. Check his [recent video](https://youtu.be/etEo6Lf7QIc) to understand. I agree and understand, looking at the results gives more satisfaction. But there is another way too.

**Reply by quasiblob** [reply]
I've tried tensorboard a couple of times, but I find it pretty hard to figure out stuff as I don't have ML background. I think I didn't get it working for TI training in A1111. 

BTW, thanks for the link, actually I did already start watching the video you linked yesterday, definitely will continue watching it as soon as I have time!

Thanks! 👍

**Reply by bmaltais** [reply]
You would need to ask kohya_ss dorectly on his repo as he is the one adding those features to the trainers: https://github.com/kohya-ss/sd-scripts

**Reply by Woisek** [reply]
> You can try tensorboard to monitor logs to understand the learning rate. Check his [recent video](https://youtu.be/etEo6Lf7QIc) to understand. 

I watched this video too, but I can't figure out how to start or setup TensorBoard. Any description on that to find?
It would be nice if Tensorboard would be integrated into the GUI for easy access.

**Reply by Eagle07-Sudo** [reply]
Me too. I can't start tensorboard.
when I open powershell or anaconda prompt and go to the kohya_ss folder and then write:
tensorboard.exe --logdir "paste log folder path"
I got ValueError: Duplicate plugins for name projector

**Reply by Woisek** [reply]
@Eagle07-Sudo 
This "issue" is almost solved, Tensorboard use will be integrated into the GUI.

**Reply by runner22k** [reply]
@Woisek Is it! how do you know?

**Reply by Woisek** [reply]
I switched to the dev branch like @bmaltais suggested [here](https://github.com/bmaltais/kohya_ss/issues/236#issuecomment-1445525105) and tried it out. It works like a charm, besides some QOL I'm missing.

**Reply by runner22k** [reply]
@Woisek That's great. need to check
